### PR TITLE
Add new functionalities including DAG and custom tasksets, support more schedulers, and improve GUI

### PR DESCRIPTION
**Pull request by CloverCho**
## New types of tasksets
* DAG (Directed Acyclic Graph): creates randomized dependencies between reactors (and reactions inside).
image

* Custom : supports user-defined tasksets written in custom LF files.
                   For the purpose of usage, we create a sample taskset which provides comparison between a scheduler with deadline and another scheduler without deadline
image

## GUI improvements
Reference: TU Dortmund's framework([link](https://github.com/tu-dortmund-ls12-rt/SSSEvaluation))

image

Basically, this PR also includes GUI improvements.
User can choose 
* more schedulers to evaluate
* more types of tasksets: basic, DAG, or custom
* several properties for constructing the tasksets: e.g., number of min/max workers and random seed.

When the user determines properties and clicks the run button, the program will generate several LF files,
build, execute binaries, and get information on the total physical execution time and the number of deadline misses. 
The program will also visualize physical execution time and the number of deadline misses using graphs

### CODE REVIEW

**Comment by hokeun** [comment]
@CloverCho is this PR still draft, or ready for review? If it's ready, please mark this `Ready for review`. I'd prefer to have this reviewed by @Soroosh129 before Aug 15th.

**Comment by CloverCho** [comment]
> @CloverCho is this PR still draft, or ready for review? If it's ready, please mark this `Ready for review`. I'd prefer to have this reviewed by @Soroosh129 before Aug 15th.

It is ready for review! I will change the status of this PR

File: C/src/TaskSetGenerator/tasksets/DagTaskSet.py [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+    =}
+}
+
+main reactor {
+    runner = new SimpleDagRunner(exe_time=$EXE_TIME$);
+
+    $TASKCONFIG$
+}
+        """
+        return template
```
Comment (hokeun) [inline C/src/TaskSetGenerator/tasksets/DagTaskSet.py:229]: Indentation seems off.

File: C/src/TaskSetGenerator/DagComponent.lf [hunk 2]
```diff
@@ -0,0 +1,64 @@
+/**
```
Comment (hokeun) [inline C/src/TaskSetGenerator/DagComponent.lf:1]: Does this file need to be added to the repository?

**Review by hokeun — COMMENTED** [review]
I think binary files such as BasicPlot.cpython-38.pyc, BasicPlot.cpython-38.pyc should not be checked in. Same for an empty file, __init__.py.

Also, I recommend adding a new line at the end of files to avoid these:
Screen Shot 2022-08-19 at 9 44 02 AM

File: C/src/TaskSetGenerator/DagComponent.lf (hunk 2 above)
Comment (CloverCho) [reply C/src/TaskSetGenerator/DagComponent.lf]: The file was created for individual practice, so it is not necessary.
I removed the file, as well as the file in __pycache__ folder you mentioned above.

**Review by CloverCho — COMMENTED** [review]
I reviewed this PR myself and delete some unnecessary newlines.

**Comment by lhstrh** [comment]
Closing this PR because this has moved into its own repo: https://github.com/lf-lang/taskset-generator

**Pull request closed without merging**